### PR TITLE
Start of the Greenplum dialect implementation 

### LINKED ISSUE #4373: New Greenplum dialect (based off Postgres)
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

A new dialect specific to VMWare Greenplum due to the differences in allowed & required keywords from stock Postgres: https://docs.vmware.com/en/VMware-Tanzu-Greenplum/6/greenplum-database/GUID-ref_guide-feature_summary.html

### Use case

My team wants to be able to use this tool but it fails to parse or incorrectly flags/allows keywords that are unique to a Greenplum instance.  

For example, a valid CREATE TABLE statement in Greenplum should define the distribution key(s) for the system to disperse across the nodes of the system:

```
create table schema.a_new_table (
      column_1 int
    , column_2 int
    , column_3 varchar
)
distributed by (column_1);
```
The parser fails to parse this.

Additionally, certain common postgres features, like triggers, are not allowed in Greenplum, so any code attempting to create one should be caught by the tool as invalid.

### Dialect

Postgres

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by JackWolverson**
fixes #4373
this is a start of the greenplum dialect support 
it implements the distributed by keywords 
and the WITH ( <storage_parameter> = value) 

### Are there any other side effects of this change that we should be aware of?
Fixes a definition in Postgres

### Pull Request checklist
- [ x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: .github/workflows/add-issue-labels.yaml
Comment (tunetheweb) [inline .github/workflows/add-issue-labels.yaml:13]: Reminder to maintainers: need to add the `greenplum` label before merging this.

File: src/sqlfluff/dialects/dialect_greenplum.py
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_greenplum.py:5]:
```suggestion
"""The Greenplum dialect.

Greenplum (http://www.greenplum.org/) is a Massively Parallel Postgres,
so we base this dialect on Postgres.
"""
```

File: src/sqlfluff/dialects/dialect_greenplum.py
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_greenplum.py:151]: This is also different from Postgres but I think you're actually right here. Do you fancy correcting Postgres as part of this PR?

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Looks good.

A few minor comments.

File: src/sqlfluff/dialects/dialect_greenplum.py
Comment (JackWolverson) [reply src/sqlfluff/dialects/dialect_greenplum.py]: resolved in latest commit

File: src/sqlfluff/dialects/dialect_greenplum.py
Comment (JackWolverson) [reply src/sqlfluff/dialects/dialect_greenplum.py]: resolved in latest commit

File: .github/workflows/add-issue-labels.yaml
Comment (JackWolverson) [reply .github/workflows/add-issue-labels.yaml:13]: greenplum is in there ? ive added it between exasol and hive so its alphabetical

Comment (tunetheweb) [reply .github/workflows/add-issue-labels.yaml:13]: I mean add it to GitHub itself as an actual label that this action can then use.

File: src/sqlfluff/dialects/dialect_greenplum.py
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_greenplum.py]: Nah, think it's small enough (and obviously wrong enough) that it's OK. I added it as a comment above in the "Are there any other side effects of this change that we should be aware of?" section above.

**Review by tunetheweb — APPROVED** [review]
LGTM!

File: .github/workflows/add-issue-labels.yaml
Comment (tunetheweb) [reply .github/workflows/add-issue-labels.yaml:13]: Have added that to https://github.com/sqlfluff/sqlfluff/labels now.

**Pull request merged**
